pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -666,16 +672,13 @@ impl RemoteTimelineClient {
         let mut guard = self.upload_queue.lock().unwrap();
         let upload_queue = guard.initialized_mut()?;
 
-        let no_bail_here = || {
-            let with_generations =
-                self.schedule_unlinking_of_layers_from_index_part0(upload_queue, names);
+        let with_generations =
+            self.schedule_unlinking_of_layers_from_index_part0(upload_queue, &names);
 
-            self.schedule_deletion_of_unlinked0(upload_queue, with_generations);
+        self.schedule_deletion_of_unlinked0(upload_queue, with_generations);
 
-            // Launch the tasks immediately, if possible
-            self.launch_queued_tasks(upload_queue);
-        };
-        no_bail_here();
+        // Launch the tasks immediately, if possible
+        self.launch_queued_tasks(upload_queue);
```

Comment:
I had to drop the closure here when I made the `names` to be borrowed. TBH I did not want to understand the reason and if I could had worked it around.